core/src/main/java/org/ldaptive/ad/control/LazyCommitControl.java

```diff
@@ -40,6 +40,15 @@ public LazyCommitControl(final boolean critical)
 
 
   @Override
+  public boolean equals(final Object o)
+  {
+    if (o == this) {
```

Comment:
I don't think this self comparison in the control classes is warranted. I could see maybe if you had a static collection of controls that was used throughout the codebase where equality checking would actually return true, but I don't see that usage pattern anywhere. Feels like a premature optimization.